sdk/anomalydetector/ai-anomaly-detector-rest/samples-dev/detectChangePointSample.ts

```diff
@@ -0,0 +1,298 @@
+        { timestamp: new Date("2017-01-02T01:05:00Z"), value: 2526771 },
+        { timestamp: new Date("2017-01-02T01:10:00Z"), value: 2526773 },
+        { timestamp: new Date("2017-01-02T01:15:00Z"), value: 2526793 },
+        { timestamp: new Date("2017-01-02T01:20:00Z"), value: 2527010 },
+        { timestamp: new Date("2017-01-02T01:25:00Z"), value: 2527031 },
+        { timestamp: new Date("2017-01-02T01:30:00Z"), value: 3178096 },
+        { timestamp: new Date("2017-01-02T01:35:00Z"), value: 3196305 },
+        { timestamp: new Date("2017-01-02T01:40:00Z"), value: 3196202 },
+        { timestamp: new Date("2017-01-02T01:45:00Z"), value: 3196202 },
+        { timestamp: new Date("2017-01-02T01:50:00Z"), value: 3196205 },
+        { timestamp: new Date("2017-01-02T01:55:00Z"), value: 3229232 },
+        { timestamp: new Date("2017-01-02T02:00:00Z"), value: 3230463 },
+        { timestamp: new Date("2017-01-02T02:05:00Z"), value: 3230468 },
+        { timestamp: new Date("2017-01-02T02:10:00Z"), value: 3230468 },
+        { timestamp: new Date("2017-01-02T02:15:00Z"), value: 3230468 },
+        { timestamp: new Date("2017-01-02T02:20:00Z"), value: 3230470 },
+        { timestamp: new Date("2017-01-02T02:25:00Z"), value: 3230467 },
+        { timestamp: new Date("2017-01-02T02:30:00Z"), value: 3230506 },
+        { timestamp: new Date("2017-01-02T02:35:00Z"), value: 3231536 },
+        { timestamp: new Date("2017-01-02T02:40:00Z"), value: 3230723 },
+        { timestamp: new Date("2017-01-02T02:45:00Z"), value: 3230727 },
+        { timestamp: new Date("2017-01-02T02:50:00Z"), value: 3230727 },
+        { timestamp: new Date("2017-01-02T02:55:00Z"), value: 3230727 },
+        { timestamp: new Date("2017-01-02T03:00:00Z"), value: 3231880 },
+        { timestamp: new Date("2017-01-02T03:05:00Z"), value: 3232283 },
+        { timestamp: new Date("2017-01-02T03:10:00Z"), value: 3232283 },
+        { timestamp: new Date("2017-01-02T03:15:00Z"), value: 3232283 },
+        { timestamp: new Date("2017-01-02T03:20:00Z"), value: 3232325 },
+        { timestamp: new Date("2017-01-02T03:25:00Z"), value: 3232323 },
+        { timestamp: new Date("2017-01-02T03:30:00Z"), value: 3232327 },
+        { timestamp: new Date("2017-01-02T03:35:00Z"), value: 3232340 },
+        { timestamp: new Date("2017-01-02T03:40:00Z"), value: 3232713 },
+        { timestamp: new Date("2017-01-02T03:45:00Z"), value: 3232729 },
+        { timestamp: new Date("2017-01-02T03:50:00Z"), value: 3232756 },
+        { timestamp: new Date("2017-01-02T03:55:00Z"), value: 3233500 },
+        { timestamp: new Date("2017-01-02T04:00:00Z"), value: 3233500 },
+        { timestamp: new Date("2017-01-02T04:05:00Z"), value: 3233500 },
+      ],
+      stableTrendWindow: 10,
+      threshold: 0.99,
+    },
+    headers: { "Content-Type": "application/json" },
+  };
+  const result = await client.path("/timeseries/changepoint/detect").post(options);
```

Comment:
Is this a long running operation? Also, what do customers get once the operation completes? Do they get the detection result? If so, it will be interesting to show them what information they can use from the response.

Reply:
You should also show error response handling. There is a helper function that can be imported `isUnexpected` that tells you if (according to swagger) if the status code is a success or failure, and will tell Typescript which type to use for intellisense

```ts
if (isUnnexpected(result)){
   throw result.body.error;
}

// Do something with success result, users have intellisense at this point
```

Reply:
I have quoted a snippet below for this issue.